src/main/java/com/eternalcode/parcellockers/gui/ParcelListGUI.java

```diff
@@ -53,30 +60,48 @@ public void showParcelListGUI(Player player) {
-                .toList()) {
-            ParcelMeta meta = parcel.meta();
-            String sender = Bukkit.getPlayer(parcel.sender()).getName();
-            String receiver = Bukkit.getPlayer(meta.getReceiver()).getName();
-
-            gui.addItem(ItemBuilder.from(Material.CHEST_MINECART)
-                    .name(this.miniMessage.deserialize(meta.getName()))
-                    .lore(this.miniMessage.deserialize("&3UUID: ").append(Component.text(parcel.uuid().toString())),
-                            this.miniMessage.deserialize("&3Sender: ").append(Component.text(sender)),
-                            this.miniMessage.deserialize("&3Receiver: ").append(Component.text(receiver)),
-                            this.miniMessage.deserialize("&3Size: ").append(Component.text(meta.getSize().toString())),
-                            this.miniMessage.deserialize("&3Priority: ").append(Component.text(meta.isPriority() ? "&aYes" : "&cNo")),
-                            this.miniMessage.deserialize("&3Description: ").append(Component.text(meta.getDescription())),
-                            this.miniMessage.deserialize("&3Position: ").append(Component.text(meta.getDestinationLocker().getPosition().toString())),
-                            this.miniMessage.deserialize("&3Recipients: ").append(Component.text(meta.getRecipients().stream().map(Bukkit::getPlayer).map(HumanEntity::getName).toList().toString())))
-                    .flags(ItemFlag.HIDE_ATTRIBUTES)
-                    .asGuiItem());
+        this.parcelDatabaseService.findAll().whenComplete((parcels, throwable) -> {
+            for (Parcel parcel : parcels) {
+                List<String> newLore = this.replaceParcelPlaceholders(parcel, parcelItem.getItemStack().getItemMeta().getLore());
+                parcelItem.getItemStack().getItemMeta().setLore(newLore);
+
+                gui.addItem(parcelItem);
+            }
+
+            gui.setItem(49, closeItem);
+            this.server.getScheduler().runTask(this.plugin, () -> gui.open(player));
+        });
+
+    }
+
+    public List<String> replaceParcelPlaceholders(Parcel parcel, List<String> lore) {
+        if (lore == null || lore.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        ParcelMeta meta = parcel.meta();
+        List<String> newLore = new ArrayList<>();
+
+        for (String line : lore) {
+            line = line.replace("{UUID}", parcel.uuid().toString());
+            line = line.replace("{NAME}", meta.getName());
+            line = line.replace("{SENDER}", this.server.getPlayer(parcel.sender()).getName());
+            line = line.replace("{RECEIVER}", this.server.getPlayer(meta.getReceiver()).getName());
+            line = line.replace("{SIZE}", meta.getSize().toString());
+            line = line.replace("{PRIORITY}", meta.isPriority() ? ChatColor.GREEN + "Yes" : ChatColor.RED + "&cNo");
```

Comment:
Imo te ChatColor do wywalenia -> zastąp to czymś bardziej czytelnym

Reply:
Pewnie